logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationsAndConnectionsUseCase.kt

```diff
@@ -11,19 +11,21 @@ fun interface ObserveConversationsAndConnectionsUseCase {
      * Convenience UseCase that allows to get conversations and connections merged and desc sorted by date
      * Using this will allow clients not handling specific sorting or merging logic
      */
-    suspend operator fun invoke(): Flow<List<ConversationDetails>>
+    suspend operator fun invoke(): Flow<ConversationListDetails>
 }
 
 internal class ObserveConversationsAndConnectionsUseCaseImpl(
     private val observeConversationListDetailsUseCase: ObserveConversationListDetailsUseCase,
     private val observeConnectionListUseCase: ObserveConnectionListUseCase
 ) : ObserveConversationsAndConnectionsUseCase {
-    override suspend fun invoke(): Flow<List<ConversationDetails>> {
-        return combine(observeConversationListDetailsUseCase(), observeConnectionListUseCase()) { conversations, connections ->
-            (conversations.conversationList + connections).sortedWith(
+    override suspend fun invoke(): Flow<ConversationListDetails> {
+        return combine(observeConversationListDetailsUseCase(), observeConnectionListUseCase()) { conversationListDetails, connections ->
+            val sortedConversationWithConnections = (conversationListDetails.conversationList + connections).sortedWith(
                 compareByDescending<ConversationDetails> { it.conversation.lastModifiedDate }
                     .thenBy(nullsLast()) { it.conversation.name?.lowercase() }
             )
+
+            conversationListDetails.copy(conversationList = sortedConversationWithConnections)
         }.distinctUntilChanged()
     }
```

Comment:
we are using this use case in the AR I had to make it return ConversationListDetails instead of <List<ConversationDetails>> so that we can display all the needed information